### PR TITLE
Passing active status to Dropdown from Nav/isItemActive to fix #133

### LINKED ISSUE #133: Nav-bar item activate doesn't work in Dropdown
### What steps will reproduce the problem?

Open the class yii\bootstrap4\Dropdown

### What's expected?

Using $itemOptions for activate link/item in Nav-bar (and for over options).

### What do you get instead?

$itemOptions not used and in result - we can't activate menu-item.

### PR DESCRIPTION
**Pull request by Thoulah**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | yes
| Tests pass?   | yes
| Fixed issues  | #133

### CODE REVIEW

**Comment by Thoulah** [comment]
Please don't merge yet, need further testing.

Currently URL detection doesn't seem to work

**Comment by Thoulah** [comment]
This code works:
```
echo yii\bootstrap4\Nav::widget([
        'items' => [
            ['label' => 'Main', 'url' => ['/site/index']],
            [
                'label' => 'Admin',
                'items' => [
                    ['label' => 'Users', 'url' => ['/site/users'], 'active' => true],
                    ['label' => 'Roles', 'url' => ['/site/roles']],
                    ['label' => 'Statuses', 'url' => ['/site/statuses']],
                ],
            ],
        ],
]);
```

Removing `active` and let the route detection does its magic gives me nothing...
Any help would be appreciated.

**Comment by samdark** [comment]
@simialbi, @AsDonAs any ideas?

**Comment by Thoulah** [comment]
I will add this test in the evening

```    public function testActive()
    {
        $this->mockAction('site', 'users');

        Nav::$counter = 0;
        $out = Nav::widget([
            'items' => [
                [
                    'label' => 'Main',
                    'active' => ['site/index'],
                ],
                [
                    'label' => 'Admin',
                    'items' => [
				        ['label' => 'Users', 'url' => ['site/users']],
                        ['label' => 'Roles', 'url' => ['site/roles']],
                        ['label' => 'Statuses', 'url' => ['site/statuses']]
					],
                ],
            ],
        ]);

        $expected = <<<EXPECTED
<ul id="w0" class="nav"><li class="nav-item"><a class="nav-link" href="/base/index.php?r=site%2Findex">Main</a></li>
<li class="dropdown nav-item"><a class="dropdown-toggle nav-link" href="#" data-toggle="dropdown">Admin</a><div id="w1" class="dropdown-menu"><a class="dropdown-item active" href="/base/index.php?r=site%2Fusers">Users</a>
<a class="dropdown-item" href="/base/index.php?r=site%2Froles">Roles</a>
<a class="dropdown-item" href="/base/index.php?r=site%2Fstatuses">Statuses</a></div></li></ul>
EXPECTED;

        $this->assertEqualsWithoutLE($expected, $out);
        $this->removeMockedAction();
    }
```

Then the result should be matched to the expected code.

**Comment by simialbi** [comment]
I will work on this problem tomorrow. I think it's a massive overkill to add this much logic to just remove an (not really) problematic `active` class. But let's see, may be there is a simple solution.

**Comment by Thoulah** [comment]
`Dropdown` doesn't have `isItemActive` function... that's why

**Comment by Thoulah** [comment]
@samdark: Tests pass now, but I don't understand.

I changed the test condition itself and suddenly get different result. AND without changing the src it affected another test result. Is something wrong with the NavTest file?

**Comment by simialbi** [comment]
@Thoulah If you agree I'll check your code and ma

**Comment by Thoulah** [comment]
@simialbi Please do :) Better more eyes on it to make sure all is fine.

**Comment by simialbi** [comment]
I really think this pr produces more problems than it solves. I watched the code and I think e.g. it will break the `active` state of a `Nav` inside a `NavBar`. To be honest, I would revert #139 too and look for a better solution.

**Comment by Thoulah** [comment]
`NavBar` itself never calls on `Nav`, so if it works from `Nav` and from `DropDown` (like it does now), it works.
What problems do you see this PR creating? It is just passing 1 value (`(bool) $active`) down to `DropDown`, nothing more.

**Comment by simialbi** [comment]
Pls see https://github.com/Thoulah/yii2-bootstrap4/pull/1, I think this is absolutely sufficient. What do you think?

**Comment by Thoulah** [comment]
There are 4 failing tests
Also I would like to have some consistency. We have `active` option in `Nav` and therefor I prefered to copy the same behaviour in `Dropdown`. This seems prefered over having to set active class in options.

**Comment by simialbi** [comment]
Fixed
https://travis-ci.com/Thoulah/yii2-bootstrap4/builds/114594113
I understand your intention. But the use case of setting a dropdown item manually active, outside a navigation tends nearby 0. And inside a navigation the navigation class sets the active state. In my opinion it's beautiful but not necessary and a bit overkill.

**Comment by Thoulah** [comment]
Perhaps setting it manually to true or false tends to nearby 0, but from a condition is not a rare occasion (eg `'active' => date('w') === 5`).
For the seconds part, why not create clean code without useless clutter.

I only added 1 variable and 1 condition, so I don't see how your solution is much cleaner.

On the positive side, I like you chances to the tests very much and will happily take those over (in the evening, at work now), along with credits to you for your time spend on this issue :+1:
I will make a manual merge of both your changes and mine. I expect it to be very clean, with nice tests.

**Comment by Thoulah** [comment]
I think it's now very clean. If you see any possible pitfalls, please let me know.

File: tests/DropdownTest.php [hunk 1]
```diff
@@ -119,14 +119,13 @@ public function testActive()
                     [
                         'label' => 'Dropdown2',
                         'items' => [
-                            ['label' => 'Page3', 'content' => 'Page3', 'url' => '/', 'active' => true],
+                            ['label' => 'Page3', 'content' => 'Page3', 'url' => '/', 'linkOptions' => ['class' => 'active']],
```
Comment (simialbi) [inline tests/DropdownTest.php]: If you keep `active` in `Dropdown`, you should keep your version here with `'active' => true`

Comment (Thoulah) [reply tests/DropdownTest.php]: Absolutely correct! This one slipped through from copy

**Comment by Thoulah** [comment]
That's on purpose. See https://getbootstrap.com/docs/4.3/components/navs/

**Comment by simialbi** [comment]
It's not in Navbar:
https://getbootstrap.com/docs/4.3/components/navbar/#nav

**Comment by Thoulah** [comment]
Ok, did not see that, but:

> Active states—with .active—to indicate the current page can be applied directly to .nav-links **_or_** their immediate parent .nav-items.

Since we have them applied on `nav-links` already now, `nav-items` can be left as is.

File: src/Nav.php [hunk 2]
```diff
@@ -81,11 +81,6 @@ class Nav extends Widget
      * @see isItemActive
      */
     public $activateItems = true;
-    /**
-     * @var bool whether to activate the container when the menu item is active.
-     * @since 2.0.3
-     */
-    public $activateContainer = true;
```
Comment (samdark) [inline src/Nav.php]: That could be considered a backwards compatibility break. Is this property useless? If not, I'd like to keep it. If it is useless, we need to at least mention it in upgrade notes (by creating UPGRADE.md like the one in the core framework).

**Comment by samdark** [comment]
@simialbi, @Thoulah please ping me when the pull request is ready for final review/merge.

File: src/Nav.php (hunk 2 above)
Comment (Thoulah) [reply src/Nav.php]: It has become useless. I will check UPGRADE.md later today

Comment (Thoulah) [reply src/Nav.php]: Do I place the file in src, or in the root folder?

**Comment by Thoulah** [comment]
@simialbi: Was ist Ihre Meinung?

**Comment by simialbi** [comment]
@Thoulah: Oh bitte Duze mich. So alt bin ich noch nicht 😄 .
> Active states—with .active—to indicate the current page can be applied directly to .nav-links or their immediate parent .nav-items.

Das war mir nicht bekannt. Ich bin mir aber ziemlich sicher, dass das nicht immer so war. Somit müsste man evtl. im `composer.json` noch den minimalen Version Constraint von `npm-asset/bootstrap` auf die aktuelle Version `^4.3` updaten (um sicher zu gehen).

**Comment by Thoulah** [comment]
Sorry, Worte kenn ich, aber die Grammatik... ärgerlich
Vielleicht wurde Boostrap geändert, das weiß ich nicht.
Ich werde die Minimalversion ändern und fertig :+1: 

@samdark: Мы готовы

**Comment by simialbi** [comment]
Oh I'm sorry, I thought cause you're location is defined as `Germany`.
For me it's ok like this. I think it could be merged.
@samdark I think we're ready

**Comment by Thoulah** [comment]
It is Germany, but not native German

File: src/Nav.php (hunk 2 above)
Comment (samdark) [reply src/Nav.php]: root.

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**